.github/workflows/java_tests.yml

```diff
@@ -0,0 +1,160 @@
+    inputs:
+      runDataflow:
+        description: 'Type "true" if you want to run Dataflow tests (GCP variables must be configured, check CI.md)'
+        default: false
+  schedule:
+    - cron: '10 2 * * *'
+  push:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+  pull_request:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+    paths: ['sdks/java/**', 'model/**', 'runners/**', 'examples/java/**', 'examples/kotlin/**', 'release/**']
+
+
+jobs:
+
+  check_gcp_variables:
+    timeout-minutes: 5
+    name: "Check GCP variables set"
+    runs-on: ubuntu-latest
+    outputs:
+      gcp-variables-set: ${{ steps.check_gcp_variables.outputs.gcp-variables-set }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: "Check are GCP variables set"
+        run: "./scripts/ci/ci_check_are_gcp_variables_set.sh"
+        id: check_gcp_variables
+        env:
+          GCP_PROJECT_ID: ${{ secrets.GCP_PROJECT_ID }}
+          GCP_SA_EMAIL: ${{ secrets.GCP_SA_EMAIL }}
+          GCP_SA_KEY: ${{ secrets.GCP_SA_KEY }}
+          GCP_TESTING_BUCKET: ${{ secrets.GCP_TESTING_BUCKET }}
+
+  java_unit_tests:
+    name: 'Java Unit Tests'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      # :sdks:java:core:test
+      - name: Run :sdks:java:core:test
+        run: ./gradlew :sdks:java:core:test
+      - name: Upload test logs for :sdks:java:core:test
+        uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: java_unit_tests-sdks-java-core-test-${{ matrix.os }}
+          path: sdks/java/core/build/reports/tests/test/**
+      # :sdks:java:harness:test
+      - name: Run :sdks:java:harness:test
+        if: ${{ always() }}
+        run: ./gradlew :sdks:java:harness:test
+      - name: Upload test logs for :sdks:java:harness:test
+        uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: java_unit_tests-sdks-java-harness-test-${{ matrix.os }}
+          path: sdks/java/harness/build/reports/tests/test/**
+      # :runners:core-java:test
+      - name: Run :runners:core-java:test
+        if: ${{ always() }}
+        run: ./gradlew :runners:core-java:test
+      - name: Upload test logs for :runners:core-java:test
+        uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: java_unit_tests-runners-core-java-test-${{ matrix.os }}
+          path: runners/core-java/build/reports/tests/test/**
+
+  java_wordcount_direct_runner:
+    name: 'Java Wordcount Direct Runner'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Run WordCount Unix
+        shell: bash
+        run: |
```

Comment:
Why is a shell used in some jobs but not others (specified here as a config and used in the run config below)? Just curious if it is needed for some reason here specifically since it is an integration test.

Reply:
The reason behind it is to override the default powershell on windows and ensure that gradle will use bash as well. This way there is no need for separate steps for windows and unix systems.